### PR TITLE
User login

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: test/testContext.js [hunk 1]
```diff
@@ -13,8 +13,8 @@ import { db } from '@server/db/db'
 const createAdminUser = async () => {
   await db.multi(`
     -- Insert the admin user to be used in the test suite:
-    INSERT INTO "user" (name, email)
-    values ('Admin', 'admin@openforis.org')
+    INSERT INTO "user" (name, email, password, status)
+    values ('Admin', 'admin@openforis.org', 'test_password', 'ACCEPTED')
```
Comment (minotogna) [inline test/testContext.js]: I would use the ```User.userStatus.ACCEPTED```

File: core/user/user.js [hunk 2]
```diff
@@ -16,11 +17,15 @@ export const isEqual = ObjectUtils.isEqual
...
 export const getAuthGroups = ObjectUtils.getAuthGroups
 export const getPrefs = R.propOr({}, keys.prefs)
 export const hasProfilePicture = R.propEq(keys.hasProfilePicture, true)
 
+// ====== UPDATE
+export const dissocPassword = R.dissoc(keys.password)
+
 // ====== CHECK
 export const isSystemAdmin = user => user && R.any(AuthGroup.isSystemAdminGroup)(getAuthGroups(user))
 export const hasAccepted = R.pipe(getName, StringUtils.isNotBlank)
```
Comment (minotogna) [inline core/user/user.js]: this should use userStatus

File: server/modules/auth/manager/authManager.js [hunk 3]
```diff
@@ -1,3 +1 @@
 export { fetchGroupByUuid } from '../repository/authGroupRepository'
```
Comment (minotogna) [inline server/modules/auth/manager/authManager.js]: remove JWT tables

File: server/modules/user/repository/userRepository.js [hunk 4]
```diff
@@ -6,20 +6,20 @@ import * as User from '@core/user/user'
 import * as Survey from '@core/survey/survey'
 import * as AuthGroup from '@core/auth/authGroup'
 
-const selectFields = ['uuid', 'name', 'email', 'prefs']
+const selectFields = ['uuid', 'name', 'email', 'password', 'prefs']
```
Comment (minotogna) [inline server/modules/user/repository/userRepository.js]: i wouldn't always fetch the password.

File: server/modules/user/service/userService.js [hunk 5]
```diff
@@ -62,33 +60,29 @@ export const inviteUser = async (user, surveyId, surveyCycleKey, email, groupUui
...
+      // Send email
+      const msgParams = {
+        serverUrl,
+        email,
+        password,
+        surveyLabel,
+        groupLabel,
+        temporaryPasswordMsg: '$t(emails.userInvite.temporaryPasswordMsg)',
       }
+      await Mailer.sendEmail(email, 'emails.userInvite', msgParams, lang)
```
Comment (minotogna) [inline server/modules/user/service/userService.js]: test if Mailer.sendEmail fails the user is removed from db

File: server/system/middleware/sessionMiddleware.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import * as session from 'express-session'
+import connectPgSimple from 'connect-pg-simple'
+
+const db = require('@server/db/db')
+
+const pgSession = connectPgSimple(session)
+
+const sessionOptions = {
+  secret: process.env.FOO_COOKIE_SECRET,
```
Comment (minotogna) [inline server/system/middleware/sessionMiddleware.js]: add this to the README and env.template // use ProcessUtils

File: webapp/app/actions.js [hunk 7]
```diff
@@ -39,8 +27,11 @@ export const initApp = () => async dispatch => {
       survey,
     })
   } catch (error) {
-    dispatch({ type: appPropsChange, status: AppState.appStatus.ready, i18n })
-    CognitoAuth.logout()
+    dispatch({
```
Comment (minotogna) [inline webapp/app/actions.js]: how come if user is not present, there's an exception ?

File: webapp/loggedin/modules/users/userList/userListView.js [hunk 8]
```diff
@@ -77,7 +77,6 @@ const UsersListView = ({ canInvite, user, surveyInfo, history }) => {
       module={'users'}
       className="users-list"
       gridTemplateColumns={'35px repeat(3, 1fr) 10rem 50px'}
-      headerLeftComponent={UsersHeaderLeft}
       rowHeaderComponent={UsersRowHeader}
```
Comment (minotogna) [inline webapp/loggedin/modules/users/userList/userListView.js]: i would leave the UsersHeaderLeft, make sure invitation is working

File: test/testContext.js (hunk 1 above)
Comment (SteRiccio) [reply test/testContext.js]: right

File: core/user/user.js (hunk 2 above)
Comment (SteRiccio) [reply core/user/user.js]: right

File: server/modules/auth/manager/authManager.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/auth/manager/authManager.js]: ok

File: server/modules/user/repository/userRepository.js (hunk 4 above)
Comment (SteRiccio) [reply server/modules/user/repository/userRepository.js]: right

File: server/system/middleware/sessionMiddleware.js (hunk 6 above)
Comment (SteRiccio) [reply server/system/middleware/sessionMiddleware.js]: ok, but I would call the env variable something like SESSION_ID_COOKIE_SECRET (secret to sign the session ID cookie)

File: webapp/loggedin/modules/users/userList/userListView.js (hunk 8 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/users/userList/userListView.js]: ok

File: server/modules/user/service/userService.js (hunk 5 above)
Comment (SteRiccio) [reply server/modules/user/service/userService.js]: yes, Mailer.sendEmail is invoked in the transaction block, the transaction rolls back correctly

File: webapp/app/actions.js (hunk 7 above)
Comment (SteRiccio) [reply webapp/app/actions.js]: right, the error is never thrown, I'll fix this.

**Pull request merged**